src/components/Map.js

```diff
@@ -19,13 +25,27 @@ const Map = ({
     ).then((res) => setNearbyPlaces(res));
   }, [mapProperties.center.lat, mapProperties.center.lng, setNearbyPlaces]);
 
+  function handleApiLoaded(map, maps) {
+    setMapsObj(map, maps);
+  }
+
+  // only runs if maps object is populated
+  if (!isObjEmpty(mapsObj)) {
```

Comment:
```suggestion
  if (mapsObj) {
```
Utilizes the falsey nature of an empty object.

Reply:
Unfortunately, this results in the following: `TypeError: Cannot read property 'addListener' of undefined`